flight/PiOS/posix/pios_flash_posix.c

```diff
@@ -107,32 +111,23 @@ void PIOS_Flash_Posix_Destroy(uintptr_t chip_id)
 
 static int32_t PIOS_Flash_Posix_StartTransaction(uintptr_t chip_id)
 {
-#if 0
 	struct flash_posix_dev * flash_dev = (struct flash_posix_dev *)chip_id;
 
-	/* TODO: Needs to be a semaphore like other platforms.  In the meantime
-	 * neutering this does nothing bad.
-	 *
-	 * Alternatively could choose to have a counter maintained using
-	 * atomics here.
-	 */
-	assert(!flash_dev->transaction_in_progress);
-
-	flash_dev->transaction_in_progress = true;
-#endif
+	if (PIOS_Semaphore_Take(flash_dev->transaction_lock,
+				PIOS_SEMAPHORE_TIMEOUT_MAX) != true) {
```

Comment:
I wonder if timeout should be a param... that kind of thing is annoying in c since you can't have a default though. :(